src/arch/x86/kernel/smp_sys.c

```diff
@@ -126,6 +136,11 @@ VISIBLE void boot_node(void)
         fail("boot_node failed for some reason :(\n");
     }
 
+#if defined(CONFIG_KERNEL_MCS) && defined(CONFIG_DEBUG_BUILD)
+    clock_sync_test = getCurrentTime();
+    /* Make sure clock_sync_test is set before smp_aps_index */
+    asm volatile("" ::: "memory");
```

Comment:
Use `__atomic_signal_fence(__ATOMIC_ACQ_REL)`, because this is not just a compile barrier, but also a CPU/Pipeline barrier?

Reply:
Then better leave this explicit barrier out and change the comment to say: x86 guarantees global order, volatile qualifier prevents compiler reordering.

Reply:
Done.